src/foam/lib/csv/CSV.js

```diff
@@ -0,0 +1,256 @@
+        Function: function(o) { /* Ignore functions in CSV */ },
+        Object: function(o) { /* Ignore generic objects in CSV */ }
+      }, function(o, prefix, first) { this.outputHeaderTitle(o, prefix, first); })
+    },
+
+    function outputProperty(o, p, first) {
+      if ( ! this.propertyPredicate(o, p) ) return;
+      this.output(o[p.name], first);
+    },
+
+    function reset() {
+      this.buf_ = '';
+      return this;
+    },
+
+    function outputPrimitive(val, first) {
+      this.out(first ? '' : this.delimiter, val);
+      return this;
+    },
+
+    function escapeString(s) {
+      if ( s.includes(',') ) {
+        // Surrounds fields with commas in quotes
+        // Escapes inner quotes by adding another quote char (Google Sheets strategy)
+        s = '"' + s.replace('"', '""') + '"';
+      }
+      
+      return s;
+    },
+
+    {
+      name: 'output',
+      code: foam.mmethod({
+        Undefined:    function(o, first) { this.outputPrimitive(this.undefinedStr, first); },
+        Null:         function(o, first) { this.outputPrimitive(null, first); },
+        String:       function(o, first) { this.outputPrimitive(this.escapeString(o), first); },
+        AbstractEnum: function(o, first) { this.outputPrimitive(o.ordinal, first); },
+        FObject:   function(o, first) {
+          if ( o.outputCSV ) {
+            o.outputCSV(this)
+            return;
+          }
+          
+          var ps = o.cls_.getAxiomsByClass(foam.core.Property);
+
+          for ( var i = 0 ; i < ps.length ; i++ ) {
+            this.outputProperty(o, ps[i], (i == 0 && first));
+          }
+        },
+        Array:        function(o) { /* Ignore arrays in CSV */ },
+        Function:     function(o) { /* Ignore functions in CSV */ },
+        Object:       function(o) { /* Ignore generic objects in CSV */ }
+      }, function(o, first) { this.outputPrimitive(o, first); })
+    },
+
+    function fromCSV(className, s, sink) {
+      if ( ! s ) throw 'Invalid CSV input to convert. Arguments must be (className, csvString).'
+      var lines = s.split('\n');
+
+      if ( lines.length == 0 ) throw 'Insufficient CSV Input';
+
+      // Trims quotes and splits CSV row into array
+      var props = this.splitIntoValues(lines[0]);
+
+      for ( var i = 1 ; i < lines.length ; ++i ) {
+        var values = this.splitIntoValues(lines[i]);
+
+        // Skips blank lines
+        if ( values.length == 0 ) continue;
+
+        // Calls for creation of new model, and `puts` into sink
+        sink.put(this.createModel(props, values, className));
+      }
+
+      return sink;
+    },
+
+    function splitIntoValues(csvString) {
+      if ( ( csvString == undefined ) || ( csvString.length == 0) ) return [];
+
+      var parser = foam.lookup('foam.lib.csv.CSVParser').create();
+      return parser.parseString(csvString).map(field => field.value == undefined ? '' : field.value);
+    },
+
+    function createModel(props, values, className) {
```

Comment:
I tried passing in the model (via `foam.lookup('fullQualifier')`), however, it gives the following error: `customModel.create is not a function`

When I pass in just the className, `foam.lookup('fullQualifier')` is exactly how I convert it into a model. I’m not sure what’s causing the error in that case.